app/src/main/java/com/infomaniak/drive/data/cache/FileController.kt

```diff
@@ -872,6 +865,28 @@ object FileController {
         }
     }
 
+    private fun FileActivity.removeFileWhenEnteringFolder(
+        isDeletionAction: Boolean,
+        realm: Realm,
+        returnResponse: ArrayMap<Int, FileActivity>,
+        currentFolder: File,
+    ) {
+        if (returnResponse[fileId] == null || returnResponse[fileId]?.createdAt?.time == createdAt.time) { // Api fix
+            getParentFile(fileId = fileId, realm = realm)?.let { localFolder ->
+                when {
+                    isDeletionAction && localFolder.id == currentFolder.id -> {
+                        removeFile(fileId, customRealm = realm, recursive = false)
+                    }
+                    localFolder.id != file?.parentId && localFolder.id == currentFolder.id -> {
+                        updateFile(localFolder.id, realm) { it.children.remove(file) }
+                    }
+                    else -> Unit
+                }
+            }
+            returnResponse[fileId] = this
+        }
+    }
+
```

Comment:
You can keep it as before and just simplify it in case you don't want to delete it